model/riscv_sys_control.sail

```diff
@@ -153,6 +166,49 @@ function is_CSR_defined (csr : csreg, p : Privilege) -> bool =
     /* supervisor mode: address translation */
     0x180 => haveSupMode() & (p == Machine | p == Supervisor), // satp
 
+    /* supervisor CSR access */
+    0x150 => haveSupMode() & (p == Machine | p == Supervisor), // siselect
+    0x151 => {
+      if haveSupMode() & (p == Machine | p == Supervisor) & menvcfg.CDE() == 0b1 then
+          match siselect.REG() {
+            0x040 => mcounteren.CY() == 0b1,
+            0x042 => mcounteren.IR() == 0b1,
+            _ => false
+          }
+        else false
+    }, // sireg
+    0x152 => {
+      if haveSupMode() & (p == Machine | p == Supervisor) & menvcfg.CDE() == 0b1 &
+        sizeof(xlen) == 32 then
+           match siselect.REG() {
+            0x040 => mcounteren.CY() == 0b1,
+            0x042 => mcounteren.IR() == 0b1,
+            _ => false
+          }
+        else false
+    }, // sireg2
+    0x153 => false, // sireg3
+    0x155 => {
+      if haveSupMode() & (p == Machine | p == Supervisor) & menvcfg.CDE() == 0b1 then
+          match siselect.REG() {
+            0x040 => mcounteren.CY() == 0b1,
+            0x042 => mcounteren.IR() == 0b1,
+            _ => false
+          }
+        else false
+    }, // sireg4
+    0x156 => {
+      if haveSupMode() & (p == Machine | p == Supervisor) & menvcfg.CDE() == 0b1 &
+        sizeof(xlen) == 32 then
+          match siselect.REG() {
+            0x040 => mcounteren.CY() == 0b1,
+            0x042 => mcounteren.IR() == 0b1,
+            _ => false
+          }
+        else false
+    }, // sireg5
+    0x157 => false, // sireg6
```

Comment:
These checks would want to go into a `is_allowed` type function which can then include extension specific legalization based on ranges of `siselect/miselect` values. For instance, CDE only applies to Sscdeleg but not to other extensions. Here may want the more generic `mstateen` check only (mstateen check is missing).